gems/aws-sdk-core/lib/aws-sdk-core/ec2_metadata.rb

```diff
@@ -183,7 +183,7 @@ def http_put(connection, ttl)
 
     def open_connection
       uri = URI.parse(@endpoint)
-      http = Net::HTTP.new(uri.hostname || @endpoint, @port || uri.port)
+      http = Net::HTTP.new(uri.hostname || @endpoint, uri.port || @port)
```

Comment:
is `@port` used anywhere else? (ie, printed in to_s or anything?) should we set it if so?

Reply:
It doesn't appear to be. It's a direct override to the class.